src/components/Admin/AddPopUpClinic/AdminPopUpClinic.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState } from "react";
+import { Card, Form, FormLayout, TextField } from "@shopify/polaris";
+import { useTranslation } from "react-i18next";
+import { postalCodeIsValid } from "../../../utils/postalCode";
+import { phoneNumberIsValid } from "../../../utils/phoneNumber";
+
+export interface IPopUpClinic {
+  name: string;
+  date: string;
+  street: string;
+  city: string;
+  province: string;
+  postalCode: string;
+  phone: string;
+  websiteUrl: string;
+  [key: string]: string;
+}
+
+const handleSubmit = () => {
+  return true;
+};
+
+export function AdminPopUpClinic() {
+  const { t } = useTranslation();
+
+  const [popUpClinic, setPopUpClinc] = useState<IPopUpClinic>({
+    name: "",
+    date: new Date().toDateString(),
+    street: "",
+    city: "",
+    province: "",
+    postalCode: "",
+    phone: "",
+    websiteUrl: "",
+  });
+
+  const [popUpClinicErrors, setPopUpClincErrors] = useState<IPopUpClinic>({
+    name: "",
+    date: "",
+    street: "",
+    city: "",
+    province: "",
+    postalCode: "",
+    phone: "",
+    websiteUrl: "",
+  });
+
+  const onFormChanged = (field: string, value: string) => {
```

Comment:
I think it would make sense to split this function into two - `onFormChanged` should be responsible for updating state, and a second `validate` function which would ensure values are passing business rules. Then your control would look something like:

```
<TextField
  name="postalcode"
  type="text"
  error={validate("postalCode", popUpClinicErrors.postalCode)}
  value={popUpClinic.postalCode}
  label={t("postalcode")}
  onChange={(value) => onFormChanged("postalCode", value)}
/>
```